src/parser/CodecParser.cpp

```diff
@@ -0,0 +1,233 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "CodecParser.h"
+
+#include <bento4/Ap4Utils.h>
+using namespace adaptive;
+
+AdtsType CAdaptiveAdtsHeaderParser::GetAdtsType(AP4_ByteStream* stream)
+{
+  AP4_DataBuffer buffer;
+  buffer.SetDataSize(AP4_EAC3_HEADER_SIZE); // max header size is 64 (E-AC3)
+  AdtsType adtsType = AdtsType::NONE;
+
+  if (!AP4_SUCCEEDED(stream->Read(buffer.UseData(), AP4_EAC3_HEADER_SIZE)))
+    return adtsType;
+
+  AP4_BitReader bits(buffer.GetData(), AP4_EAC3_HEADER_SIZE);
+  AP4_UI32 syncWord = bits.ReadBits(16);
+  if ((syncWord & AP4_ADTS_SYNC_MASK) == AP4_ADTS_SYNC_PATTERN)
+  {
+    adtsType = AdtsType::AAC;
+  }
+  else if ((syncWord & AP4_AC4_SYNC_MASK) == AP4_AC4_SYNC_PATTERN)
+  {
+    adtsType = AdtsType::AC4;
+  }
+  else if (syncWord == AP4_AC3_SYNC_PATTERN)
+  {
+    bits.SkipBits(24);
+    AP4_UI32 bitStreamID = bits.ReadBits(5);
+    if ((bitStreamID > 10) && (bitStreamID <= 16))
+    {
+      adtsType = AdtsType::EAC3;
+    }
+    else if (bitStreamID <= 10)
+    {
+      adtsType = AdtsType::AC3;
+    }
+  }
+  AP4_Position currentPos;
+  stream->Tell(currentPos);
+  stream->Seek(currentPos - (AP4_EAC3_HEADER_SIZE));
+  return adtsType;
+}
+
+AP4_Result CAdaptiveAdtsParser::FindFrameHeader(AP4_AacFrame& frame)
+{
+  unsigned int available;

```

Comment:
this var `unsigned int available;`
is declared but unused also on other method below, please delete it

all other changes ok
just remember to make a separate commit for bento ver bump